docs/api/modules/Internationalization.md

```diff
@@ -0,0 +1,209 @@
+[👈 Return to Overview](../API.md)
+
+# Internationalization
+
+To enable internationalization modules can contain a locale directory in the root of
+the module. Messages for each locale should be stored in JSON files within
+the locale directory with filenames using the [BCP-47](https://tools.ietf.org/html/bcp47) formatted language tag
+(e.g. en-US.json).
+
+**Contents**
+* [Locale Structure](#locale-structure)
+  * [Environment Specific Data](#environment-specific-data)
+  * [ONE_CLIENT_LOCALE_FILENAME](#ONE_CLIENT_LOCALE_FILENAME)
+* [Loading Language Packs](#loading-language-packs)
+  * [`loadLanguagePack`](#loadLanguagePack)
+  * [`queryLanguagePack`](#queryLanguagePack)
+  * [`updateLocale`](#updateLocale)
+  * [`getLocalePack`](#getLocalePack)
+
+
+## Locale Structure
+
+Following the below structure will allow [`one-app-locale-bundler`](https://github.com/americanexpress/one-app-cli/tree/master/packages/one-app-locale-bundler) to correctly
+bundle your locale files.
+
+Module Root:
+```
+module
+├── README.md
+├── locale
+|   └── en-US.json
+|   └── es-MX.json
+├── package.json
+└── src
+    └── index.js
+```
+
+This will result in:
+```
+build
+└── 1.0.0
+    ├── <moduleName>.browser.js
+    ├── <moduleName>.legacy.browser.js
+    ├── <moduleName>.node.js
+    ├── en-us
+    |   ├── <moduleName>.json
+    |   ├── integration.json
+    |   └── qa.json
+    └── es-mx
+        ├── <moduleName>.json
+        ├── integration.json
+        └── qa.json
+```
+
+> 💬 For use with [React Intl](https://github.com/formatjs/react-intl) the language pack needs
+> to be a flat JSON object and will not work with a nested structure.
+
+Example `en-US.json`:
+```JSON
+{
+  "loading": "Loading...",
+  "welcome": "welcome to my module"
+}
+```
+
+### Environment Specific Data
+
+[`one-app-locale-bundler`](https://github.com/americanexpress/one-app-cli/tree/master/packages/one-app-locale-bundler) also allows your module to provide environment
+specific data. Currently One App supports three different environments;
+production, qa and integration (development).
+
+The directory name is what defines the key the data goes under,
+below is an example of how you could include "link" specific data.
```

Comment:
How does the `links` differ from a `lang-pack` and how would I load it (use it) in my module? Would it be done the same way as in `cultured-frankie` for example?

https://github.com/americanexpress/one-app/blob/master/prod-sample/sample-modules/cultured-frankie/0.0.0/src/components/CulturedFrankie.jsx#L81

If I am using `links` would I need to create a `copy.json` instead of using just using environments?
https://github.com/americanexpress/one-app-cli/tree/master/packages/one-app-locale-bundler#-usage

And is there a specific format for `links`?

Reply:
these are going to be part of the Recipes documentation